akka-actor/src/main/scala/akka/io/dns/DnsSettings.scala

```diff
@@ -0,0 +1,162 @@
+
+  private def isInetAddress(name: String): Boolean =
+    ipv4Address.findAllMatchIn(name).nonEmpty || ipv6Address.findAllMatchIn(name).nonEmpty
+
+  // Note that the corresponding man page doesn't actually dictate the format of this field,
+  // just the keywords and their meanings. See http://man7.org/linux/man-pages/man5/resolv.conf.5.html
+  private[io] val NameserverLine = """^\s*nameserver\s+(.*)$""".r
+
+  // OS specific. No encoding or charset is specified by the man page as I recall.
+  // See http://man7.org/linux/man-pages/man5/resolv.conf.5.html.
+  def parseSystemNameServers(resolvConf: File): Option[immutable.Seq[InetSocketAddress]] =
+    try {
+      val addresses =
+        for {
+          line ← Source.fromFile(resolvConf).getLines()
+          addr ← NameserverLine.findFirstMatchIn(line).map(_.group(1))
+        } yield parseNameserverAddress(addr)
+      Some(addresses.toList)
+    } catch {
+      case NonFatal(_) ⇒ Option.empty
+    }
+
+  def failUnableToDetermineDefaultNameservers =
+    throw new IllegalStateException("Unable to obtain default nameservers from JNDI or via reflection. " +
+      "Please set `akka.io.dns.async.nameservers` explicitly in order to be able to resolve domain names.")
+
+}
+
+object DnsSettings {
+
+  private final val DnsFallbackPort = 53
+
+  /**
+   * INTERNAL API
+   * Find out the default search domains that Java would use normally, e.g. when using InetAddress to resolve domains.
+   *
+   * The default nameservers are attempted to be obtained from: jndi-dns and from `sun.net.dnsResolverConfiguration`
+   * as a fallback (which is expected to fail though when running on JDK9+ due to the module encapsulation of sun packages).
+   *
+   * Based on: https://github.com/netty/netty/blob/4.1/resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsServerAddressStreamProvider.java#L58-L146
+   */
+  private[akka] def getDefaultSearchDomains(): Try[List[InetSocketAddress]] = {
```

Comment:
this crazy dance is directly based on Netty's impl.
Without it we would not know which nameservers to use.
The akka-dns impl did not find the nameservers and always forced users to set them explcitly.

Netty also falls back to the Google DNS by default if the JNDI and reflection one can not be found. We could consider doing the same.

Reply:
I.e. `akka.io.dns.async.nameservers-fallback` whichw e'd set by default t the google ones